test/e2e/runner/rpc.go

```diff
@@ -182,3 +195,35 @@ func waitForNode(ctx context.Context, node *e2e.Node, height int64) (*rpctypes.R
 		}
 	}
 }
+
+// getLatestBlock returns the last block that all active nodes in the network have
+// agreed upon i.e. the earlist of each nodes latest block
+func getLatestBlock(ctx context.Context, testnet *e2e.Testnet) (*types.Block, error) {
+	var earliestBlock *types.Block
+	for _, node := range testnet.Nodes {
+		// skip nodes that don't have state or haven't started yet
+		if node.Stateless() {
+			continue
+		}
+		if !node.HasStarted {
+			continue
+		}
+
+		client, err := node.Client()
+		if err != nil {
+			return nil, err
+		}
+
+		wctx, cancel := context.WithTimeout(ctx, 10*time.Second)
```

Comment:
Is there some reason we're adding a timeout here? It seems like it affects the semantics of this helper.

Maybe add:

```go
// This will fail if any started node that has state fails to respond within 10 seconds.
```

Or maybe just pass in a timeout to the helper?

Reply:
I believe there is a timeout attached to the entire operation. This timeout is just "don't fail the test or block forever if a node blips for a few seconds.

Eventually all nodes have to get to the specified height, so I think the semantics are safe.

Reply:
It's not that I think it's unsafe—more that it seems likely to fail spuriously. But maybe 10 seconds is a long time in this context? It wasn't clear.